pkg/util/log/slog/wrapper.go

```diff
@@ -0,0 +1,215 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package slog provides an slog implementation of the LoggerInterface interface.
+package slog
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"runtime"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log/types"
+)
+
+var _ types.LoggerInterface = (*Wrapper)(nil)
+
+const baseStackDepth = 4
+
+// Wrapper is a wrapper around the slog.Handler interface.
+// It implements the LoggerInterface interface.
+type Wrapper struct {
+	handler slog.Handler
+	flush   func()
+	close   func()
+
+	attrs           []slog.Attr
+	extraStackDepth int
```

Comment:
so we're supporting negative values for this? Possibly we should just have the stack depth be configured in a declarative way instead of having this option be an adjustment to the base value constant

Reply:
The reason I've done it this way is to stay compatible with seelog, I'll most likely change it once we drop seelog altogether 😄 
Seelog's `LoggerInterface` has a `SetAdditionalStackDepth` method, which.. sets an additional stack depth
In particular it can be called at any point basically (and every function in `pkg/util/log` which takes a stack depth does so, they call `SetAdditionalStackDepth`, then log, then call `SetAdditionalStackDepth` again to revert to the previous value)
So implementing it like this allows being 100% compatible with seelog, which makes things waaay easier for now